weather_stats/docker-compose.yml

```diff
@@ -4,21 +4,44 @@ services:
     build:
       context: ./weatherstatsapi
       dockerfile: Dockerfile
-    command: uvicorn weather.app:app --host 0.0.0.0 --port 5052
+    command: uvicorn weather.app:app --host 0.0.0.0 --port 5053
     env_file: .env
+    platform: linux/amd64
     ports:
-      - "5052:5052"
+      - "5053:5053"
     depends_on:
       - cache
+    networks:
+      hfnet:
+        ipv4_address: 192.168.0.3
   cache:
     image: redis:7.0-bullseye
+    networks:
+      hfnet:
+        ipv4_address: 192.168.0.2
+    platform: linux/amd64
+    networks:
+      - hfnet
     restart: always
     ports:
       - "6379:6379"
     command: redis-server --loglevel warning --requirepass redispw
   app:
+    platform: linux/amd64
+    networks:
+      - hfnet
     build:
       context: ./weatherstatsapp
       dockerfile: Dockerfile
     ports:
       - "8080:80"
+
+
+networks:
+  hfnet:
```

Comment:
Can you tell me something more about this?

Reply:
This is to hardcode the IP's of the containers because the IP of some services are hardcoded inside the applications because those addresses are hardcoded in the golem-compose file. 

Without these, you risk that the container is spun up on another IP address and then nginx can't find the service because it expects a service at an hardcoded address.

Reply:
So this essentially solves that problem.